src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -31,0 +1,94 @@
+import { Todo } from '../../types/Todo';
+import { useState, useEffect } from 'react';
+import { Option } from '../Option';
+
+interface Props {
+  allTodos?: Todo[];
+  setTodos: (value: Todo[]) => void;
+}
+
+type FilterType = 'all' | 'active' | 'completed';
+
+export const TodoFilter: React.FC<Props> = ({ allTodos, setTodos }) => {
+  const [inputSearch, setInputSearch] = useState<string>('');
+  const [close, setClose] = useState(false);
+  const [filter, setFilter] = useState<FilterType>('all');
+
+  useEffect(() => {
+    if (allTodos) {
+      let newTodos = [...allTodos];
+
+      if (filter === 'active') {
+        newTodos = allTodos?.filter(todo => !todo.completed);
+      } else if (filter === 'completed') {
+        newTodos = allTodos?.filter(todo => todo.completed);
```

Comment:
The use of optional chaining `allTodos?.filter` is unnecessary here because `allTodos` is already checked for existence. You can safely use `allTodos.filter`.